### ISSUE TITLE
[BUG] Run workspace package script failed if script has chained commands(&&)

### ISSUE THREAD

**Issue by fpsqdb** [npm/cli#2990]
### Current Behavior:

Execute `npm init` to initialize a workspace package named `package1`.
Execute `npm run test -w package1`

```
PS D:\temp\temp4\root> npm run test -w package1

> package1@1.0.0 test
> echo "Error: no test specified" && exit 1

"Error: no test specified"
npm ERR! Lifecycle script `test` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: package1@1.0.0
npm ERR!   at location: D:\temp\temp4\root\package1
PS D:\temp\temp4\root>
```

Works fine if excute `npm run --prefix package1 test`
```
PS D:\temp\temp4\root> npm run --prefix package1 test

> package1@1.0.0 test
> echo "Error: no test specified" && exit 1

"Error: no test specified"
PS D:\temp\temp4\root>
```

### Expected Behavior:

Run without error.

### Steps To Reproduce:
```.
root
  ├── package.json ->  { "workspaces": ["package1"] }
  └── package1
      └── package.json { "scripts": { "test": "echo \"Error: no test specified\" && exit 1" } }
```
```
npm run test -w package1
```

### Environment:

- OS: Win10 19042.867
- Node: 12.20.0
- npm: 7.7.6

**Comment by ljharb** [comment]
Doesn't `npm run --prefix package1 test` run the test script from the current directory, but with the cwd set the `package1`?

What happens if you cd into package1 and run `npm test`? I'd assume the exact same thing.

**Comment by fpsqdb** [comment]
All the command run from `root` directory not `package1`.
run `npm test` from `package1` works fine.

I have edited the issue to make it clearer

**Comment by fpsqdb** [comment]
And another.
`package1->package.json`
```  json
"scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "test2": "echo \"test2\"",
    "start": "npm run test2"
  }
```
workes fine
```
PS D:\temp\temp4\root> npm run test2 -w package1

> package1@1.0.0 test2
> echo "test2"

"test2"
PS D:\temp\temp4\root>
```
error
```
PS D:\temp\temp4\root> npm run start -w package1

> package1@1.0.0 start
> npm run test2

npm ERR! No workspaces found:
npm ERR!   --workspace=package1

npm ERR! A complete log of this run can be found in:
npm ERR!     D:\Cache\npm-cache\_logs\2021-03-31T03_15_06_747Z-debug.log
npm ERR! Lifecycle script `start` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: package1@1.0.0
npm ERR!   at location: D:\temp\temp4\root\package1
PS D:\temp\temp4\root>
```

**Comment by lukekarrys** [comment]
I was running into this issue in a project as well and I think it is resolved in `npm@7.8.0`.

> 
> error
> 
> ```
> PS D:\temp\temp4\root> npm run start -w package1
> 
> > package1@1.0.0 start
> > npm run test2
> 
> npm ERR! No workspaces found:
> npm ERR!   --workspace=package1
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     D:\Cache\npm-cache\_logs\2021-03-31T03_15_06_747Z-debug.log
> npm ERR! Lifecycle script `start` failed with error:
> npm ERR! Error: command failed
> npm ERR!   in workspace: package1@1.0.0
> npm ERR!   at location: D:\temp\temp4\root\package1
> PS D:\temp\temp4\root>
> ```

As this example showed, I think the issue is with running nested scripts within a workspace. So in this case `npm run start` runs `npm run test2` inside `package1/`. I created a minimal repo that reproduces this with `npm@7.7.6` but the command runs successfully with `npm@7.8.0`. https://github.com/lukekarrys/npm-7.7.6-issue-2990

**Comment by wraithgar** [comment]
Looks like this is fixed, as per @lukekarrys's research
